code/datums/components/crafting/crafting.dm

```diff
@@ -198,6 +198,12 @@
 	var/list/contents = get_surroundings(crafter, recipe.blacklist)
 	var/send_feedback = 1
 	var/dest_turf = get_turf(crafter)
+
+	if(isopenturf(dest_turf) && !(recipe.crafting_flags & CRAFT_CHECK_DENSITY))
```

Comment:
why are you checking the density flag instead of the turf flag? I mean I assume most of these recipies have both flags, but still